pageserver/src/http/routes.rs

```diff
@@ -224,22 +224,21 @@ async fn timeline_detail_handler(request: Request<Body>) -> Result<Response<Body
             info_span!("timeline_detail_handler", tenant = %tenant_id, timeline = %timeline_id)
                 .entered();
         let repo = tenant_mgr::get_repository_for_tenant(tenant_id)?;
-        Ok::<_, anyhow::Error>(match repo.get_timeline(timeline_id)?.local_timeline() {
-            None => TimelineInfo::Remote {
-                timeline_id,
-                tenant_id,
-            },
-            Some(timeline) => TimelineInfo::Local {
+        let local_timeline = {
+            // TODO (current) handle other timeline states
```

Comment:
Also interesting to know, how do we want to deal with `timeline_attach_handler` http requests, that are synchronous at their current definition, yet the S3 sync machinery is now very async and does not give any latency guarantees.

Reply:
I think there is no need to list remote timelines at all. In our cloud setup console has all the knowledge about tenants and timelines. Requirement to know about all remote timelines on each pageserver is quite hard to implement correctly because this involves global (even distributed) state. Result of the list call can quite quickly become out of sync and hence cannot provide correct answer to the question "what remote timelines are currently available". That means that we need to introduce some background polling to update the inmemory index, and even with that 404 answer on timeline detail can mean two completely different things. It is either permanent absence of the timeline or it was just not synced yet. This leads to possible retries from clients, timeout management and all the ~~delights~~ consequences of eventual consistency. So I think we should avoid that coordination as much as possible or use separate metadata service for discovery/management of this state. See https://github.com/zenithdb/rfcs/pull/16 for more details on metadata service approach. 

> how do we want to deal with timeline_attach_handler http requests

I think the way with that is how it is done in tenant relocation test: https://github.com/zenithdb/zenith/pull/995/files#diff-28d93314ea7a547530e9c963649b70dedbbabe0cc9cb60e07f797f432cd75f6cR196. We call attach, and then wait by polling timeline detail for status to become local. This can also change with the presence of the metadata server. When pageserver can advertise statuses of the timelines it knows about. Metadata service can also provide a way to subscribe for the changes. (AFAIK both etcd and consul support that)

Reply:
Sure, got the idea of no remote sync state.

> I think the way with that is how it is done in tenant relocation test

I was mainly confused by the fact that there we somehow store that timeline id within the same test scope, while in reality, without listing the remote timeline ids, there could be obvious way to recover id we need to use.
But if the metadata server is present, presumably that should also be solved by it.

Reply:
> there could be obvious way to recover id we need to use

Yeah, either metadata service or console should help with that. But I agree that this is not obvious, and we should think about what to do for self hosted installations